src/protocol/textDocument.ts

```diff
@@ -358,6 +358,11 @@ export interface TextDocumentRegistrationOptions {
      * the document selector provided on the client side will be used.
      */
     documentSelector: DocumentSelector | null
+
+    /**
+     * User-visible name of the of the provider being registered.
+     */
+    providerName: string
```

Comment:
Based on how you're using it elsewhere, this should be optional (ie `providerName?: string`)

Reply:
Nope, it should be required. Otherwise, it will show up as blank in the provider selector. Added a check to guarantee this where the type is asserted.